cvpy/tests/test_cvat_project.py

```diff
@@ -149,62 +158,74 @@ def test_cvat_project_post_images(self):
-        cas_connection.image.loadimages(labellevels=5,
-                                        casout=cas_table_encoded,
-                                        caslib='dlib',
-                                        decode=False)
-
-        cas_connection.addcaslib(name='bigdata',
-                                 activeOnAdd=False,
-                                 path='/bigdisk/lax/patela/data',
-                                 dataSource='PATH',
-                                 subdirectories=True)
+        cas_table_encoded = self.cas_connection.CASTable('cas_table_encoded_save_test')
+        self.cas_connection.image.loadimages(path='images',
+                                             labellevels=5,
+                                             casout=cas_table_encoded,
+                                             caslib=self.caslib_name,
+                                             decode=False)
 
         # Create a CVATProject.
         url = TestCVATProject.cvat_url
 
-        project_name = 'MyDemoProject'
+        project_name = 'MyDemoProject_SaveTest'
         annotation_type = AnnotationType.OBJECT_DETECTION
 
         mountain_label = AnnotationLabel(name='Mountain', color='orange')
         person_label = AnnotationLabel(name='Person', color='green')
 
         labels = [mountain_label, person_label]
 
-        credentials = Credentials(self.cvat_username, self.cvat_password)
-        cvat_project = CVATProject(url=url, cas_connection=cas_connection, credentials=credentials,
+        credentials = Credentials(TestCVATProject.cvat_username, TestCVATProject.cvat_password)
+        cvat_project = CVATProject(url=url, cas_connection=self.cas_connection, credentials=credentials,
                                    project_name=project_name, annotation_type=annotation_type,
                                    labels=labels)
 
         # Post the images to the CVATProject.
         image_table_encoded = ImageTable(cas_table_encoded)
         cvat_project.post_images(image_table_encoded)
 
-        # Note: CVAT will throw an internal server error exception if we attempt to access the tasks
-        #       data to soon after an upload. So we have a sleep here to prevent the issue.
-        time.sleep(5)
+        # Save the project
+        cvat_project.save(self.caslib_name, 'cvpy', replace=True)
+
+    def test_cvat_project_resume(self):
+
+        # Resume a test project
+        cvat_project = CVATProject.resume(project_name='MyDemoProject', cas_connection=self.cas_connection,
+                                          caslib=self.caslib_name, relative_path='cvpy')
+
+        # Verify all project attributes are set correctly
+        assert cvat_project.project_version == 1
+        assert cvat_project.project_name == 'MyDemoProject'
+        assert cvat_project.annotation_type == AnnotationType.OBJECT_DETECTION
+        assert cvat_project.credentials.username is None
```

Comment:
Given that these are both supposed to be "None", does that mean we will need to create a new Credentials object after resuming a project and before continuing to do work?

Reply:
Yes, you are correct. The user will need to create a Credentials object before continuing to use the resumed project. It is similar to how you would re-authenticate to access/resume a Windows Remote Desktop session.